### PR TITLE
bump k8s deps to v1.23.9

### PR DESCRIPTION
**Pull request by Huang-Wei**
Bump k8s deps to the latest v1.23 patch before cutting a v1.23 release.

### CODE REVIEW

**Comment by Huang-Wei** [comment]
@seanmalloy could you please take a look?

File: hack/install-envtest.sh [hunk 1]
```diff
@@ -26,9 +26,9 @@ set -o pipefail
 SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 source "${SCRIPT_ROOT}/hack/lib/init.sh"
 
-# setup-envtest requires the version to be in the form of X.XX.X (e.g. 1.23.3)
+# setup-envtest requires the version to be in the form of X.Y(.Z) (e.g. 1.23 or 1.23.3)
 # thus we want to remove the 'v' from the version extracted out of the go.mod file
-version=$(cat ${SCRIPT_ROOT}/go.mod | grep 'k8s.io/kubernetes' | grep -v '=>' | awk '{ print $2 }' | sed 's/v//')
+version=$(cat ${SCRIPT_ROOT}/go.mod | grep 'k8s.io/kubernetes' | grep -v '=>' | awk '{print $NF}' | awk 'BEGIN{FS=OFS="."}NF--' | sed 's/v//')
```
Comment (Huang-Wei) [inline hack/install-envtest.sh:31]: We used to use x.y.z to refer to the exact kubebuilder-tools version, but kubebuilder-tools doesn't build every patch version. So turn to x.y to be more tolerant on version matching.

**Review by yibozhuang — APPROVED** [review]

**Comment by denkensk** [comment]
/lgtm
/approve

Thanks. @Huang-Wei

**Pull request merged**